src/components/select/utils/with-filter.hoc.js

```diff
@@ -21,7 +21,7 @@ const withFilter = (WrappedComponent) => {
 
         if (!filteredElements) {
           const noResultsText = I18n.t('select.no_results', {
-            defaultValue: 'No results'
+            defaultValue: noResultsMessage
```

Comment:
I noticed the old implementation used to have a translation for no_results_for_term, is this no longer supported by the DLS?

Also, in this conversation https://github.com/Sage/carbon/pull/2913/files#r448837015
we chose to allow placeholder to override the translation. We should be doing the same here.
In this case, if they have a translation provided globally, there is no way to override a specific instance.

Reply:
yup, you're right. Changed that to `no_results_for_term` and moved override out of the translation

Reply:
It actually has both, do we need "no results" and "new results for term"? I think "no results" is displayed if you type a value, then delete it.

Reply:
if there you delete the text then all values are displayed so we don't need that here

Reply:
OK no problem.